Firestore/core/src/firebase/firestore/util/delayed_constructor.h

```diff
@@ -121,7 +113,16 @@ class DelayedConstructor {
   }
 
  private:
-  typename std::aligned_storage<sizeof(T), alignof(T)>::type space_;
+  union Space {
+    /** Default constructor does nothing. */
+    Space() {
+    }
+    ~Space() {
+      value.~T();
+    }
+    char empty;
+    T value;
```

Comment:
I can't comment on the places you haven't changed yet, but there are few other things that should go with this change:

  * All the placement `new` invocations in this file (`new (&space_) T`) should now reference `space_.value`.
  * All the `reinterpret_cast<...>(&space_)` can become just `space_.value`.

Reply:
I've pushed a change containing these suggestions.